test/Microsoft.Diagnostics.EventFlow.Outputs.Tests/HttpOutputTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Diagnostics.EventFlow.Configuration;
+using Microsoft.Diagnostics.EventFlow.TestHelpers;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Diagnostics.EventFlow.Outputs.Tests
+{
+    public class HttpOutputTests
+    {
+        [Fact]
+        public void VerifyAllConfigOptionsMappedToConfigObject()
+        {
+            var pipelineConfigObj = new Dictionary<string, object>
+            {
+                ["outputs"] = new List<Dictionary<string, object>>
+                {
+                    new Dictionary<string, object>
+                    {
+                        ["type"] = "Http",
+                        ["format"] = "json-lines",
+                        ["ServiceUri"] = "http://localhost:1000",
+                        ["basicAuthenticationUserName"] = "mywebuser",
+                        ["basicAuthenticationUserPassword"] = "mywebpass",
+                        ["httpContentType"] = "application/x-custom",
+                    }
+                }
+            };
+
+            using (var configFile = new TemporaryFile())
+            {
+                var pipelineConfig = JsonConvert.SerializeObject(pipelineConfigObj);
+
+                configFile.Write(pipelineConfig);
+                var configBuilder = new ConfigurationBuilder();
+                configBuilder.AddJsonFile(configFile.FilePath);
+                var configuration = configBuilder.Build();
+                var outputConfigSection = configuration.GetSection("outputs");
+                var configFragments = outputConfigSection.GetChildren().ToList();
+
+                var httpOutputConfiguration = new HttpOutputConfiguration();
+                configFragments[0].Bind(httpOutputConfiguration);
+
+                Assert.Equal(httpOutputConfiguration.ServiceUri, "http://localhost:1000");
+                Assert.Equal(httpOutputConfiguration.Format, "json-lines");
+                Assert.Equal(httpOutputConfiguration.BasicAuthenticationUserName, "mywebuser");
+                Assert.Equal(httpOutputConfiguration.BasicAuthenticationUserPassword, "mywebpass");
+                Assert.Equal(httpOutputConfiguration.HttpContentType, "application/x-custom");
+            }
+        }
```

Comment:
I'd like to see a test that verifies the output will produce different data (json vs json-lines), depending on the configuration option. 

This should not be too hard if you add HttpClient as a constructor parameter. In the output code you would create an instance if null (default) is passed. In the test code you would mock it using Moq

Reply:
Hm.. I'm just a C# beginner and maybe I need some help with this goal.